.github/workflows/ci.yml

```diff
@@ -0,0 +1,72 @@
+# This workflow will do a clean install of the dependencies and run tests across different versions
+#
+# Requires scripts:
+# - bin/ci.sh
+# - bin/ci-check.sh
+
+name: elixir / main ci
+
+on:
+  push:
+    branches: [master, main]
+  workflow_dispatch:
+
+jobs:
+  precheck:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Use Elixir
+        uses: actions/setup-elixir@v1
+        with:
+          otp-version: '22.2'
+          elixir-version: '1.10.0'
+
+      - name: Install project dependencies
+        run: mix deps.get
+
+      - name: Build project
+        run: mix
+
+      - name: Run prechecks
+        run: bin/ci-check.sh
+
+      - name: Find trailing whitespace
+        uses: harupy/find-trailing-whitespace@master
+
+  ci:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        elixir: ['1.6.0', '1.7.0']
+        otp: ['19.0']
```

Comment:
These generate a matrix right? then 47-55 are the specific additions since the earlier versions don't support those versions of OTP, correct?

Reply:
Correct. Looks like Travis specified things a bit different, so I just used the same combinations. Might be good to decide what our baseline is.